Dodger Lab MoveLeft Bounds Issue

### Canvas Link

https://learning.flatironschool.com/courses/6956/assignments/239271?module_item_id=562278

### Concern

We do still have a problem, though. Even though we're only going one pixel at a time, eventually our dodger will zoom (well, relatively speaking) right out of view.

How can we prevent this? We need to check where the left edge of the dodger is and only move it if it hasn't yet reached the left edge of the game field.

We're doing essentially the same thing, but we first ensure that the dodger's left edge has not reached the left edge of its container.

Now try moving the dodger past the left edge. No can do!

Finally, implement the code needed to prevent the dodger from escaping off the right-hand side. How can we check whether the right edge of the dodger has reached the right edge of the game field? (Keep in mind that the dodger is 40px wide.)

RIGHT IS WORKING.

LEFT FAILS, When I cannot move it to the left of 0. But when I can, it passes.
TESTBED (npm test -> indexTest.js) IS WRONG. Somehow the test class is wrong and I didn't modify it. I commented the code:

//dodger.style.bottom = "0px";
//dodger.style.left = "0px";

THEN THE LEFT PASSED.


### Additional Context


  moveDodgerLeft()
    1) moves the DODGER to the left

  moveDodgerRight()
    ✓ moves the DODGER to the right


  1 passing (315ms)
  1 failing

  1) moveDodgerLeft()
       moves the DODGER to the left:

      AssertionError: expected +0 to be below +0
      + expected - actual


      at Context.<anonymous> (test/indexTest.js:17:31)
      at process.processImmediate (node:internal/timers:476:21)


### Suggested Changes

Fix the test. Then regrade all of the assignments. I think the logic is just backwards for said test. Testbed fails when starting at the left corner.

--- a/test/indexTest.js
+++ b/test/indexTest.js
@@ -13,6 +13,7 @@ describe('moveDodgerLeft()', () => {
 
     let newPosition = dodger.style.left
     newPosition = parseInt(newPosition)
+
     if (left > 0){
       expect(newPosition).to.be.below(left)
     } else {
@@ -34,6 +35,7 @@ describe('moveDodgerRight()', () => {
 
     let newPosition = dodger.style.left
     newPosition = parseInt(newPosition)
+    
     if (left < 360){
       expect(newPosition).to.be.above(left)
     } else{

--- a/test/indexTest.js
+++ b/test/indexTest.js
@@ -6,15 +6,18 @@ describe('moveDodgerLeft()', () => {
   })
 
   it('moves the DODGER to the left', () => {
-    var left = dodger.style.left
+    let left = dodger.style.left
     left = parseInt(left)
 
     moveDodgerLeft()
 
-    var newPosition = dodger.style.left
+    let newPosition = dodger.style.left
     newPosition = parseInt(newPosition)
-
-    expect(newPosition).to.be.below(left)
+    if (left > 0){
+      expect(newPosition).to.be.below(left)
+    } else {
+      expect(newPosition).to.equal(left)
+    }
   })
 })
 
@@ -24,14 +27,18 @@ describe('moveDodgerRight()', () => {
   })
 
   it('moves the DODGER to the right', () => {
-    var left = dodger.style.left
+    let left = dodger.style.left
     left = parseInt(left)
 
     moveDodgerRight()
 
-    var newPosition = dodger.style.left
+    let newPosition = dodger.style.left
     newPosition = parseInt(newPosition)
-
-    expect(newPosition).to.be.above(left)
+    if (left < 360){
+      expect(newPosition).to.be.above(left)
+    } else{
+      expect(newPosition).to.equal(left)
+    }
+    
   })
 })

